pkg/multicluster/register/registrant.go

```diff
@@ -362,8 +379,8 @@ func (c *clusterRegistrant) DeregisterCluster(
 		return err
 	}
 
-	kcSecretObjMeta := kubeconfig.SecretObjMeta(opts.Namespace, opts.ClusterName)
-	kubeClusterObjMeta := kubeClusterObjMeta(kcSecretObjMeta.Name, kcSecretObjMeta.Namespace)
+	kcSecretObjMeta := kubeconfig.SecretObjMeta(opts.Namespace, opts.ClusterName, nil)
+	kubeClusterObjMeta := kubeClusterObjMeta(kcSecretObjMeta.Name, kcSecretObjMeta.Namespace, nil)
```

Comment:
why `nil`?

Reply:
This constructs object metas for the purposes of deregistration, so we don't need the resource labels, only the name/namespace